### PR TITLE
ROX-23361: add kubernetes listener

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description

ROX-23361: add kubernetes listener for compliance remediations.

The dispatcher is currently disabled, it will be enabled as soon as Central can handle the messages send by Sensor (PR https://github.com/stackrox/stackrox/pull/10753)

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

 - Run Sensor and see that events from Kubernetes are picked up
 - Extensive testing happens with central pipeline integration, component is disabled by default

### CODE REVIEW

File: sensor/kubernetes/listener/resources/dispatcher.go
Comment (dashrews78) [inline sensor/kubernetes/listener/resources/dispatcher.go:17]: Before we merge this it should be determined if the refactor PR is going to happen or not.  If it isn't, we should put the remediation dispatcher with the rest of them.

**Review by dashrews78 — CHANGES_REQUESTED** [review]
We need to put the dispatchers in a consistent place before approval.  Everything else looks good.

**Comment by SimonBaeumer** [comment]
@dashrews78 Now they are consistent, merged PR to move the listeners: https://github.com/stackrox/stackrox/pull/10327

File: sensor/kubernetes/listener/resources/dispatcher.go
Comment (SimonBaeumer) [reply sensor/kubernetes/listener/resources/dispatcher.go]: Now they are consistent, merged the PR to move the listeners: https://github.com/stackrox/stackrox/pull/10327

**Review by lvalerom — COMMENTED** [review]
LGTM. I would either merge #10327 or move the dispatcher to `listener/resources/complianceoperator/dispatchers`

**Review by lvalerom — APPROVED** [review]
Looks good!

**Review by dashrews78 — APPROVED** [review]

**Pull request merged**
